clarity/evaluator/haspi/eb.py

```diff
@@ -1457,176 +1792,247 @@ def bm_covary(xBM, yBM, segsize, fsamp):
+            if (ref_mean_square > small) and (proc_mean_squared > small):
+                signal_cross_covariance[k, n] = unbiased_cross_correlation / np.sqrt(
+                    ref_mean_square * proc_mean_squared
+                )  # Normalized cross-covariance
             else:
-                sigcov[k, n] = 0.0
+                signal_cross_covariance[k, n] = 0.0
 
-            sigMSx[k, n] = MSx  # Save the reference MS level
-            sigMSy[k, n] = MSy  # Save the reference MS level
+            reference_mean_square[k, n] = ref_mean_square  # Save the reference MS level
+            processed_mean_square[
+                k, n
+            ] = proc_mean_squared  # Save the reference MS level
 
         # The last (half) windowed segment
         nstart = nstart + nhalf
         nstop = nstart + nhalf
-        segx = x[nstart:nstop] * window[0:nhalf]  # Window the reference
-        segy = y[nstart:nstop] * window[0:nhalf]  # Window the processed signal
-        segx = segx - np.mean(segx)  # Make 0-mean
-        segy = segy - np.mean(segy)
-        MSx = np.sum(segx**2) * halfsum2  # Normalize signal MS value by the window
-        MSy = np.sum(segy**2) * halfsum2
-
-        c = np.correlate(segx, segy, "full")
-        c = c[int(len(segx) - 1 - maxlag) : int(maxlag + len(segx))]
-
-        Mxy = np.max(np.abs(c * halfcorr))  # Unbiased cross-correlation
-        if (MSx > small) and (MSy > small):
-            sigcov[k, nseg - 1] = Mxy / np.sqrt(
-                MSx * MSy
+        reference_seg = x[nstart:nstop] * window[0:nhalf]  # Window the reference
+        processed_seg = y[nstart:nstop] * window[0:nhalf]  # Window the processed signal
+        reference_seg = reference_seg - np.mean(reference_seg)  # Make 0-mean
+        processed_seg = processed_seg - np.mean(processed_seg)
+        ref_mean_square = (
+            np.sum(reference_seg**2) * halfsum2
+        )  # Normalize signal MS value by the window
+        proc_mean_squared = np.sum(processed_seg**2) * halfsum2
+
+        correlation = np.correlate(reference_seg, processed_seg, "full")
+        correlation = correlation[
+            int(len(reference_seg) - 1 - maxlag) : int(maxlag + len(reference_seg))
+        ]
+
+        unbiased_cross_correlation = np.max(
+            np.abs(correlation * halfcorr)
+        )  # Unbiased cross-correlation
+        if (ref_mean_square > small) and (proc_mean_squared > small):
+            signal_cross_covariance[k, nseg - 1] = unbiased_cross_correlation / np.sqrt(
+                ref_mean_square * proc_mean_squared
             )  # Normalized cross-covariance
         else:
-            sigcov[k, nseg - 1] = 0.0
+            signal_cross_covariance[k, nseg - 1] = 0.0
 
-        sigMSx[k, nseg - 1] = MSx  # Save the reference MS level
-        sigMSy[k, nseg - 1] = MSy  # Save the reference MS level
+        reference_mean_square[
+            k, nseg - 1
+        ] = ref_mean_square  # Save the reference MS level
```

Comment:
Suggest having a single "# Save the reference MS level" comment before the two lines.